### ISSUE TITLE
Game stays on black screen after Save and Quit, error logs indicate “Cannot read field "field_1690" because "this.client" is null”

### ISSUE THREAD

**Issue by tariqk** [KingContaria/FastQuit#51]
[link to specific error here](https://gist.github.com/tariqk/2f27880e88891c0915fe23bcf4fbaac6)

Generally speaking, game crashes when attempting to Save and Exit, and screen stays black with no response. Switching applications and checking the log provides the above log error. Only way to resolve appears to be just to Kill the app using Prism Launcher.

Running on Minecraft 1.20.1, on Fabric 0.16.5. 

Modpack is based on [Additive 1.29.1+1.20.fabric](https://modrinth.com/modpack/additive/version/1.29.1+1.20.1.fabric), with a lot of added mods. The list of mods being used are as follows:

```
- amecs 1.3.10+mc.1.20.1
	   \-- amecsapi 1.4.0+mc1.20-pre1
	- animatica 0.6.1+1.20.4
	- architectury 9.2.14
	- athena 3.1.2
	- backported_wolves 1.1.0-1.19.3-1.20.4
	- badoptimizations 2.1.4
	- bellsandwhistles 0.4.5
	- bettergrass 1.2.1+fabric.1.20.1
	- botania 1.20.1-446-FABRIC
	   |-- fiber 0.23.0-2
	   |-- reach-entity-attributes 2.4.0
	   \-- step-height-entity-attribute 1.2.0
	- botarium 2.3.4
	   \-- team_reborn_energy 3.0.0
	- build_tools 1.0.1
	- calcmod fabric-1.20-1.3.0
	   \-- clientarguments 1.5.1
	- capes 1.5.2+1.20
	- carryon 2.1.2.7
	- cerbons_api 1.1.0
	   |-- cardinal-components-base 5.2.2
	   \-- cardinal-components-world 5.2.2
	- chipped 3.0.6
	- chisel 1.8.0
	- citresewn 1.1.5+1.20.1
	   \-- citresewn-defaults 1.1.5+1.20.1
	- cloth-config 11.1.136
	   \-- cloth-basic-math 0.6.1
	- coalnugget 1.7.1.8
	- comforts 6.4.0+1.20.1
	- configured 2.2.3
	- continuity 3.0.0-beta.5+1.20.1
	- copycats 2.1.4+mc.1.20.1-fabric
	- cpm 0.6.19a
	- create 0.5.1-f-build.1417+mc1.20.1
	   |-- com_google_code_findbugs_jsr305 3.0.2
	   |-- flywheel 0.6.10-2
	   |-- forgeconfigapiport 8.0.0
	   |-- milk 1.2.60
	   |    \-- dripstone_fluid_lib 3.0.2
	   |-- porting_lib_tags 3.0
	   |-- reach-entity-attributes 2.4.0
	   \-- registrate-fabric 1.3.62-MC1.20.1
	- create_armor_trims 2.0.1
	- create_copper_and_zinc 1.4.0
	- create_dd 0.1d
	   \-- dsbg 1.0
	- create_enchantment_industry 1.2.16
	   \-- create_dragon_lib 1.4.2
	- create_high_pressure 1.1.0
	- create_new_age 1.1.2
	- create_nuke 1.0.0
	- create_oxidized 0.1.1+1.20.1
	   \-- terraform-wood-api-v1 7.0.1
	- create_power_loader 1.5.3-mc1.20.1-fabric
	- create_structures 1.1.0
	- create_unbreakable 1.2+fabric-1.20.1
	- createaddition 1.2.4
	- createcobblestone 1.4-beta-2+fabric-1.20.1-50
	- createcontraptionterminals 1.0.0
	- createdeco 2.0.2-1.20.1-fabric
	- createdieselgenerators 2.1.4
	- createfood 1.1.2
	- createframed 1.4.6+1.20.1
	- createutilities 0.3.0+1.20.1
	   \-- team_reborn_energy 3.0.0
	- crystal_clear 0.0
	- cubes-without-borders 2.1.1+1.20
	- cull-less-leaves 1.3.0
	- design_decor 0.4.0b_fabric+1.20.1
	- dynamic_fps 3.7.3
	   \-- net_lostluma_battery 1.1.0
	- ears 1.4.6
	- enhancedblockentities 0.9+1.20
	   |-- advanced_runtime_resource_pack 0.6.7
	   \-- spruceui 5.0.0+1.20
	- entity_model_features 2.2.6
	- entity_sound_features 0.4
	- entity_texture_features 6.2.5
	   \-- org_apache_httpcomponents_httpmime 4.5.10
	- entityculling 1.7.0
	- essential-container 1.0.0
	   \-- essential-loader 1.2.3
	- estrogen 4.2.10+1.20.1-fabric
	   |-- baubly 1.0.1
	   \-- jukeboxfix 1.0.0+1.20.1
	- explorerscompass 1.20.1-2.2.3-fabric
	- extendedgears 2.1.1-1.20.1-0.5.1.f-fabric
	- extra_trinkets 1.1
	- fabric-api 0.92.2+1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-api-lookup-api-v1 1.6.36+1802ada577
	   |-- fabric-biome-api-v1 13.0.13+1802ada577
	   |-- fabric-block-api-v1 1.0.11+1802ada577
	   |-- fabric-block-view-api-v2 1.0.1+1802ada577
	   |-- fabric-blockrenderlayer-v1 1.1.41+1802ada577
	   |-- fabric-client-tags-api-v1 1.1.2+1802ada577
	   |-- fabric-command-api-v1 1.2.34+f71b366f77
	   |-- fabric-command-api-v2 2.2.13+1802ada577
	   |-- fabric-commands-v0 0.2.51+df3654b377
	   |-- fabric-containers-v0 0.1.64+df3654b377
	   |-- fabric-content-registries-v0 4.0.11+1802ada577
	   |-- fabric-convention-tags-v1 1.5.5+1802ada577
	   |-- fabric-crash-report-info-v1 0.2.19+1802ada577
	   |-- fabric-data-attachment-api-v1 1.0.0+de0fd6d177
	   |-- fabric-data-generation-api-v1 12.3.4+1802ada577
	   |-- fabric-dimensions-v1 2.1.54+1802ada577
	   |-- fabric-entity-events-v1 1.6.0+1c78457f77
	   |-- fabric-events-interaction-v0 0.6.2+1802ada577
	   |-- fabric-events-lifecycle-v0 0.2.63+df3654b377
	   |-- fabric-game-rule-api-v1 1.0.40+1802ada577
	   |-- fabric-item-api-v1 2.1.28+1802ada577
	   |-- fabric-item-group-api-v1 4.0.12+1802ada577
	   |-- fabric-key-binding-api-v1 1.0.37+1802ada577
	   |-- fabric-keybindings-v0 0.2.35+df3654b377
	   |-- fabric-lifecycle-events-v1 2.2.22+1802ada577
	   |-- fabric-loot-api-v2 1.2.1+1802ada577
	   |-- fabric-loot-tables-v1 1.1.45+9e7660c677
	   |-- fabric-message-api-v1 5.1.9+1802ada577
	   |-- fabric-mining-level-api-v1 2.1.50+1802ada577
	   |-- fabric-model-loading-api-v1 1.0.3+1802ada577
	   |-- fabric-models-v0 0.4.2+9386d8a777
	   |-- fabric-networking-api-v1 1.3.11+1802ada577
	   |-- fabric-networking-v0 0.3.51+df3654b377
	   |-- fabric-object-builder-api-v1 11.1.3+1802ada577
	   |-- fabric-particles-v1 1.1.2+1802ada577
	   |-- fabric-recipe-api-v1 1.0.21+1802ada577
	   |-- fabric-registry-sync-v0 2.3.3+1802ada577
	   |-- fabric-renderer-api-v1 3.2.1+1802ada577
	   |-- fabric-renderer-indigo 1.5.2+85287f9f77
	   |-- fabric-renderer-registries-v1 3.2.46+df3654b377
	   |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
	   |-- fabric-rendering-v0 1.1.49+df3654b377
	   |-- fabric-rendering-v1 3.0.8+1802ada577
	   |-- fabric-resource-conditions-api-v1 2.3.8+1802ada577
	   |-- fabric-resource-loader-v0 0.11.10+1802ada577
	   |-- fabric-screen-api-v1 2.0.8+1802ada577
	   |-- fabric-screen-handler-api-v1 1.3.30+1802ada577
	   |-- fabric-sound-api-v1 1.0.13+1802ada577
	   |-- fabric-transfer-api-v1 3.3.5+8dd72ea377
	   \-- fabric-transitive-access-wideners-v1 4.3.1+1802ada577
	- fabric-language-kotlin 1.12.2+kotlin.2.0.20
	   |-- org_jetbrains_kotlin_kotlin-reflect 2.0.20
	   |-- org_jetbrains_kotlin_kotlin-stdlib 2.0.20
	   |-- org_jetbrains_kotlin_kotlin-stdlib-jdk7 2.0.20
	   |-- org_jetbrains_kotlin_kotlin-stdlib-jdk8 2.0.20
	   |-- org_jetbrains_kotlinx_atomicfu-jvm 0.25.0
	   |-- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.9.0
	   |-- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.9.0
	   |-- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.6.1
	   |-- org_jetbrains_kotlinx_kotlinx-io-bytestring-jvm 0.5.4
	   |-- org_jetbrains_kotlinx_kotlinx-io-core-jvm 0.5.4
	   |-- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.7.2
	   |-- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.7.2
	   \-- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.7.2
	- fabricloader 0.16.5
	   \-- mixinextras 0.4.1
	- fabricskyboxes 0.7.3+mc1.20.1
	- farmersdelight 1.20.1-2.1.6+refabricated
	   |-- mm 2.3
	   |-- porting_lib_accessors 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_base 2.3.4+1.20.1
	   |    |-- porting_lib_attributes 2.3.4+1.20.1
	   |    |-- porting_lib_common 2.3.4+1.20.1
	   |    |-- porting_lib_core 2.3.4+1.20.1
	   |    |-- porting_lib_entity 2.3.4+1.20.1
	   |    |-- porting_lib_extensions 2.3.4+1.20.1
	   |    |-- porting_lib_fluids 2.3.4+1.20.1
	   |    |-- porting_lib_mixin_extensions 2.3.4+1.20.1
	   |    |-- porting_lib_transfer 2.3.4+1.20.1
	   |    |-- porting_lib_utility 2.3.4+1.20.1
	   |    \-- reach-entity-attributes 2.4.0
	   |-- porting_lib_config 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_lazy_registration 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_loot 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_networking 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_recipe_book_categories 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   \-- porting_lib_tool_actions 2.3.4+1.20.1
	        \-- porting_lib_core 2.3.4+1.20.1
	- fastback 0.15.6+1.20.1-fabric
	   |-- com_googlecode_javaewah_javaewah 1.2.3
	   |-- fabric-permissions-api-v0 0.2-SNAPSHOT
	   |-- net_i2p_crypto_eddsa 0.3.0
	   |-- org_apache_sshd_sshd-common 2.10.0
	   |-- org_apache_sshd_sshd-core 2.10.0
	   |-- org_eclipse_jgit_org_eclipse_jgit 6.7.0.202309050840-r
	   |-- org_eclipse_jgit_org_eclipse_jgit_ssh_apache 6.7.0.202309050840-r
	   |-- packet_tweaker 0.4.0+1.19.4
	   \-- server_translations_api 2.0.0+1.20
	        \-- packet_tweaker 0.4.0+1.19.4
	- fastquit 3.0.0+1.20+
	- ferritecore 6.0.1
	- fix-keyboard-on-linux 1.0.1
	- fsb-interop 1.3.6+mc1.20.1-build.50
	- fusion 1.1.1
	- garnished 1.9.8+1.20.1-fabric
	   \-- terraform-wood-api-v1 7.0.1
	- gravestones 1.0.9-1.20.1
	- harvest 1.3.1
	- harvestableflowers 1.1.0+1.20.1
	- immediatelyfast 1.2.21+1.20.4
	   \-- net_lenni0451_reflect 1.3.4
	- immersive_aircraft 1.1.2+1.20.1
	   \-- org_mariuszgromada_math_mathparser_org-mxparser 5.2.1
	- indium 1.0.34+mc1.20.1
	- interiors 0.5.6
	- inventoryprofilesnext 1.10.10
	- iris 1.7.2+mc1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-key-binding-api-v1 1.0.37+1802ada577
	   |-- io_github_douira_glsl-transformer 2.0.1
	   |-- org_anarres_jcpp 1.4.14
	   \-- org_antlr_antlr4-runtime 4.13.1
	- irisflw 1.1.3a
	   \-- flywheel 0.6.10-2
	- isxander-main-menu-credits 1.1.2
	- jade 11.12.0+fabric
	- java 17
	- jeed 1.20-2.2.2
	- jei 15.19.5.99
	- jeresources 1.4.0.247
	- journeymap 5.10.2
	   \-- journeymap-api-fabric 1.20-1.9-fabric-SNAPSHOT
	- just_enough_beacons 1.2.0
	- justenoughbreeding 1.4.0
	- justenoughprofessions 3.0.1
	- keybindsgaloreplus 1.1.0+1.20.4
	- lambdynlights 2.3.2+1.20.1
	   |-- pride 1.2.0+1.19.4
	   \-- spruceui 5.0.0+1.20
	- languagereload 1.6.1+1.20.1
	- leaves_us_in_peace 1.4.0
	- libipn 4.0.2
	- lithium 0.11.2
	- macos_input_fixes 1.7
	- malilib 0.16.3
	- man_of_many_planes 0.2.0+1.20.1
	- mcwbridges 3.0.0
	- mcwdoors 1.1.1
	- mcwfences 1.1.2
	- mcwfurnitures 3.3.0
	- mcwholidays 1.1.0
	- mcwlights 1.1.1
	- mcwpaintings 1.0.5
	- mcwpaths 1.0.5
	- mcwroofs 2.3.1
	- mcwtrpdoors 1.1.3
	- mcwwindows 2.3.0
	- minecraft 1.20.1
	- minihud 0.27.1
	- modelfix 1.15
	- modernfix 5.19.4+mc1.20.1
	- modmenu 7.2.2
	- molten_metals 1.20.1-0.1.4
	- moonlight 1.20-2.13.2
	- mousetweaks 2.26
	- mr_backported_wolvesterralithcompat 1.0
	- mr_create_decoadditions 1.3
	- naturescompass 1.20.1-2.2.3-fabric
	- nets 0.3.5.5
	- noisium 2.3.0+mc1.20-1.20.1
	- optigui 2.3.0-beta.5+1.20
	   |-- org_apache_commons_commons-text 1.11.0
	   \-- org_ini4j_ini4j 0.5.4
	- paintings_backport 2.0.0
	- patchouli 1.20.1-84-FABRIC
	   \-- fiber 0.23.0-2
	- playerplates 3.0.1+mc1.20.1
	- pneumonocore 1.1.4+1.20.1
	- polymorph 0.49.7+1.20.1
	   |-- cardinal-components-block 5.2.1
	   |-- cardinal-components-item 5.2.1
	   \-- spectrelib 0.13.17+1.20.1
	        |-- com_electronwill_night-config_core 3.6.7
	        \-- com_electronwill_night-config_toml 3.6.7
	- polytone 1.20-2.3.2
	   \-- net_objecthunter_exp4j 0.4.8
	- porting_lib 2.3.4+1.20.1
	   |-- porting_lib_asm 2.3.4+1.20.1
	   |-- porting_lib_attributes 2.3.4+1.20.1
	   |-- porting_lib_blocks 2.3.4+1.20.1
	   |-- porting_lib_brewing 2.3.4+1.20.1
	   |-- porting_lib_chunk_loading 2.3.4+1.20.1
	   |-- porting_lib_client_events 2.3.4+1.20.1
	   |-- porting_lib_common 2.3.4+1.20.1
	   |-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_data 2.3.4+1.20.1
	   |-- porting_lib_entity 2.3.4+1.20.1
	   |-- porting_lib_extensions 2.3.4+1.20.1
	   |-- porting_lib_fluids 2.3.4+1.20.1
	   |-- porting_lib_gametest 2.3.4+1.20.1
	   |-- porting_lib_gui_utils 2.3.4+1.20.1
	   |-- porting_lib_items 2.3.4+1.20.1
	   |-- porting_lib_level_events 2.3.4+1.20.1
	   |-- porting_lib_mixin_extensions 2.3.4+1.20.1
	   |-- porting_lib_model_builders 2.3.4+1.20.1
	   |-- porting_lib_model_generators 2.3.4+1.20.1
	   |-- porting_lib_model_loader 2.3.4+1.20.1
	   |-- porting_lib_model_materials 2.3.4+1.20.1
	   |-- porting_lib_models 2.3.4+1.20.1
	   |-- porting_lib_obj_loader 2.3.4+1.20.1
	   |-- porting_lib_registries 2.3.4+1.20.1
	   |-- porting_lib_transfer 2.3.4+1.20.1
	   |-- porting_lib_utility 2.3.4+1.20.1
	   \-- reach-entity-attributes 2.4.0
	- puzzle 1.5.2+1.20
	   |-- midnightlib 1.4.1
	   |-- puzzle-base 1.5.2+1.20
	   |-- puzzle-gui 1.5.2+1.20
	   |-- puzzle-models 1.5.2+1.20
	   \-- puzzle-splashscreen 1.5.2+1.20
	- railways 1.6.4+fabric-mc1.20.1
	   \-- mm 2.3
	- rechiseled 1.1.6
	- reeses-sodium-options 1.7.2+mc1.20.1-build.101
	- resolutioncontrol 1.20-3.0.0
	- resourcefullib 2.1.29
	   |-- com_teamresourceful_bytecodecs 1.0.2
	   \-- com_teamresourceful_yabn 1.0.3
	- shearablevines 1.1.0
	- sliceanddice 3.3.1
	   \-- forgeconfigapiport 8.0.0
	- sodium 0.5.11+mc1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-block-view-api-v2 1.0.1+1802ada577
	   |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
	   \-- fabric-resource-loader-v0 0.11.10+1802ada577
	- sodium-extra 0.5.4+mc1.20.1-build.115
	   |-- caffeineconfig 1.3.0+1.17
	   \-- crowdin-translate 1.4+1.19.3
	- sophisticatedbackpacks 3.19.5+mc1.20.1-SNAPSHOT-build.104
	   \-- cardinal-components-entity 5.2.2
	- sophisticatedcore 0.5.109+mc1.20.1-SNAPSHOT-build.104
	- sophisticatedstorage 0.8.60+mc1.20.1-SNAPSHOT-build.104
	- supermartijn642configlib 1.1.8+a
	- supermartijn642corelib 1.1.17+a
	- terralith 2.5.4
	- threadtweak 1.20.2-0.1.1
	- toms_storage 1.6.8
	- trade_cycling 1.20.1-1.0.14
	- trinkets 3.7.2
	- tweakeroo 0.17.1
	- yeetusexperimentus 2.3.0-build.4+mc1.20.1
	- yet_another_config_lib_v3 3.5.0+1.20.1-fabric
	   |-- com_twelvemonkeys_common_common-image 3.10.0
	   |-- com_twelvemonkeys_common_common-io 3.10.0
	   |-- com_twelvemonkeys_common_common-lang 3.10.0
	   |-- com_twelvemonkeys_imageio_imageio-core 3.10.0
	   |-- com_twelvemonkeys_imageio_imageio-metadata 3.10.0
	   |-- com_twelvemonkeys_imageio_imageio-webp 3.10.0
	   |-- org_quiltmc_parsers_gson 0.2.1
	   \-- org_quiltmc_parsers_json 0.2.1
	- yosbr 0.1.2
	- zoomify 2.14.0+1.20.1
	   \-- com_akuleshov7_ktoml-core-jvm 0.5.1
```

**Comment by KingContaria** [comment]
Seems to be an incompatibility with Fast Backups, I will try reproducing it with only FastQuit and Fast Backups (+ dependencies).
https://github.com/KingContaria/FastQuit/issues/50 also uses Fast Backups, so this might be related

**Comment by KingContaria** [comment]
Okay I'm fairly sure this is an incompatibility between Fast Backups and the "Allow multiple running Worlds" option, could you confirm a couple things for me?
1. The issue also happens with only FastQuit, Fast Backups and their necessary dependencies installed.
2. The issue does not happen with FastQuits "Allow multiple running Worlds" option turned off.
If this fixes the issue i will add Fast Backups to the list of mods that automatically disable this option in the next release.

**Comment by tariqk** [comment]
I can't confirm the first one yet, but I actually disabled “Allow multiple running Worlds” after seeing #50, and now, instead of crashing after “Save and Quit“, it'll crash if I quit too quickly after Saving and Exiting into the Main Menu screen. Waiting a few seconds and letting FastQuit resolve appears to not cause the problem, although it will occur if I accidentally quit the title screen too fast.

I'll update you once I test this out with a clean 0.16.5 Fabric install, using both FastBackup and FastQuit (and their deps).

**Comment by tariqk** [comment]
Right. So interestingly, without running `/backup init`, it quits fairly cleanly without any errors.

After running `/backup init` and then `/backup local`, the game exits successfully (`Process exited with code 0.`), but this error appears just before:

```
[15:04:58] [Server thread/ERROR]: Shutdown action failed.
java.lang.NullPointerException: Cannot read field "field_1755" because "this.client" is null
	at net.pcal.fastback.mod.fabric.FabricClientProvider.setMessageScreenText(FabricClientProvider.java:97) ~[fastback-0.15.6+1.20.1-fabric.jar:?]
	at net.pcal.fastback.mod.ModImpl.setMessageScreenText(ModImpl.java:98) ~[fastback-0.15.6+1.20.1-fabric.jar:?]
	at net.pcal.fastback.mod.ModImpl.onWorldStop(ModImpl.java:202) ~[fastback-0.15.6+1.20.1-fabric.jar:?]
	at net.pcal.fastback.mod.fabric.FabricClientInitializer.lambda$onInitializeClient$3(FabricClientInitializer.java:67) ~[fastback-0.15.6+1.20.1-fabric.jar:?]
	at net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents.lambda$static$6(ServerLifecycleEvents.java:76) ~[fabric-lifecycle-events-v1-2.2.2+1802ada577-15099849194dcb7e.jar:?]
	at net.minecraft.server.MinecraftServer.handler$zeg000$fabric-lifecycle-events-v1$afterShutdownServer(MinecraftServer.java:2352) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3782(MinecraftServer.java:619) ~[client-intermediary.jar:?]
	at net.minecraft.class_1132.method_3782(class_1132.java:209) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:703) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
[15:04:58] [Server thread/INFO]: [FastQuit] Finished saving "Test World" (2s)
```

This is running a fresh Minecraft instance using Prism Launcher (flatpak on on Pop!_OS 22.04 LTS, same as the last machine), Minecraft 1.20.1, using the 0.16.5 Fabric loader, with the following mod loadout:

```Loading 70 mods:
	- cloth-config 11.1.136
	   \-- cloth-basic-math 0.6.1
	- fabric-api 0.92.2+1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-api-lookup-api-v1 1.6.36+1802ada577
	   |-- fabric-biome-api-v1 13.0.13+1802ada577
	   |-- fabric-block-api-v1 1.0.11+1802ada577
	   |-- fabric-block-view-api-v2 1.0.1+1802ada577
	   |-- fabric-blockrenderlayer-v1 1.1.41+1802ada577
	   |-- fabric-client-tags-api-v1 1.1.2+1802ada577
	   |-- fabric-command-api-v1 1.2.34+f71b366f77
	   |-- fabric-command-api-v2 2.2.13+1802ada577
	   |-- fabric-commands-v0 0.2.51+df3654b377
	   |-- fabric-containers-v0 0.1.64+df3654b377
	   |-- fabric-content-registries-v0 4.0.11+1802ada577
	   |-- fabric-convention-tags-v1 1.5.5+1802ada577
	   |-- fabric-crash-report-info-v1 0.2.19+1802ada577
	   |-- fabric-data-attachment-api-v1 1.0.0+de0fd6d177
	   |-- fabric-data-generation-api-v1 12.3.4+1802ada577
	   |-- fabric-dimensions-v1 2.1.54+1802ada577
	   |-- fabric-entity-events-v1 1.6.0+1c78457f77
	   |-- fabric-events-interaction-v0 0.6.2+1802ada577
	   |-- fabric-events-lifecycle-v0 0.2.63+df3654b377
	   |-- fabric-game-rule-api-v1 1.0.40+1802ada577
	   |-- fabric-item-api-v1 2.1.28+1802ada577
	   |-- fabric-item-group-api-v1 4.0.12+1802ada577
	   |-- fabric-key-binding-api-v1 1.0.37+1802ada577
	   |-- fabric-keybindings-v0 0.2.35+df3654b377
	   |-- fabric-lifecycle-events-v1 2.2.22+1802ada577
	   |-- fabric-loot-api-v2 1.2.1+1802ada577
	   |-- fabric-loot-tables-v1 1.1.45+9e7660c677
	   |-- fabric-message-api-v1 5.1.9+1802ada577
	   |-- fabric-mining-level-api-v1 2.1.50+1802ada577
	   |-- fabric-model-loading-api-v1 1.0.3+1802ada577
	   |-- fabric-models-v0 0.4.2+9386d8a777
	   |-- fabric-networking-api-v1 1.3.11+1802ada577
	   |-- fabric-networking-v0 0.3.51+df3654b377
	   |-- fabric-object-builder-api-v1 11.1.3+1802ada577
	   |-- fabric-particles-v1 1.1.2+1802ada577
	   |-- fabric-recipe-api-v1 1.0.21+1802ada577
	   |-- fabric-registry-sync-v0 2.3.3+1802ada577
	   |-- fabric-renderer-api-v1 3.2.1+1802ada577
	   |-- fabric-renderer-indigo 1.5.2+85287f9f77
	   |-- fabric-renderer-registries-v1 3.2.46+df3654b377
	   |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
	   |-- fabric-rendering-v0 1.1.49+df3654b377
	   |-- fabric-rendering-v1 3.0.8+1802ada577
	   |-- fabric-resource-conditions-api-v1 2.3.8+1802ada577
	   |-- fabric-resource-loader-v0 0.11.10+1802ada577
	   |-- fabric-screen-api-v1 2.0.8+1802ada577
	   |-- fabric-screen-handler-api-v1 1.3.30+1802ada577
	   |-- fabric-sound-api-v1 1.0.13+1802ada577
	   |-- fabric-transfer-api-v1 3.3.5+8dd72ea377
	   \-- fabric-transitive-access-wideners-v1 4.3.1+1802ada577
	- fabricloader 0.16.5
	   \-- mixinextras 0.4.1
	- fastback 0.15.6+1.20.1-fabric
	   |-- com_googlecode_javaewah_javaewah 1.2.3
	   |-- fabric-permissions-api-v0 0.2-SNAPSHOT
	   |-- net_i2p_crypto_eddsa 0.3.0
	   |-- org_apache_sshd_sshd-common 2.10.0
	   |-- org_apache_sshd_sshd-core 2.10.0
	   |-- org_eclipse_jgit_org_eclipse_jgit 6.7.0.202309050840-r
	   |-- org_eclipse_jgit_org_eclipse_jgit_ssh_apache 6.7.0.202309050840-r
	   |-- packet_tweaker 0.4.0+1.19.4
	   \-- server_translations_api 2.0.0+1.20
	        \-- packet_tweaker 0.4.0+1.19.4
	- fastquit 3.0.0+1.20+
	- java 17
	- minecraft 1.20.1```

So basically just FastQuit, FastBackup and all their dependencies only.

**Comment by KingContaria** [comment]
This is caused by Fast Backups setting it's client field to null on shutdown, but since FastQuit is still saving the world at that time it causes some logic to fail. This can only be fixed by Fast Backups and actually was fixed during the update to 1.20.4: https://github.com/pcal43/fastback/commit/61640d380b3a8dcef09e6b95a074870a3951fc78#diff-8d1fb9aeecfabca170ddd0bf935e3dec28bdfc4090cdfbd1a75be1c033efe1a7R97
I can't do much about it and it's a fairly rare issue in an old version of Minecraft, so i doubt this will be fixed

**Comment by tariqk** [comment]
got it. thanks for the update! at least there's still a workaround with regards to this, (i.e. just waiting for FastQuit to complete before opening another world or quitting Minecraft).
